cli/src/pcluster/models/cluster.py

```diff
@@ -186,7 +179,13 @@ def _persist_stack_resources(self, keys):
     def _update_template(self, template):
         """Update template of the running stack according to updated template."""
         try:
-            AWSApi.instance().cfn.update_stack(self.name, template, self._params)
+            bucket = S3BucketFactory.get_cluster_bucket(
```

Comment:
After merging #2605 let's move this code specific for the bucket management in the Cluster, before calling the `stack.update_template()`

Reply:
done